tools/topology/topology2/cavs/platform/intel/hdmi-generic.conf

```diff
@@ -0,0 +1,152 @@
+Object.Dai {
+	HDA.4 {
+		name iDisp1
+		id $HDMI1_ID
+		default_hw_conf_id 1
+		Object.Base.hw_config.HDA4 {}
+		direction playback
+	}
+	HDA.5 {
+		name iDisp2
+		id $HDMI2_ID
+		default_hw_conf_id 2
+		Object.Base.hw_config.HDA5 {}
+		direction playback
+	}
+	HDA.6 {
+		name iDisp3
+		id $HDMI3_ID
+		default_hw_conf_id 3
+		Object.Base.hw_config.HDA6 {}
+		direction playback
+	}
+}
+Object.Pipeline {
+	passthrough-playback.20 {
+		Object.Widget.pipeline.1 {
+			stream_name $HDMI1_PIPELINE_STREAM_NAME
+		}
+		Object.Widget.copier.1 {
+			stream_name $HDMI1_PCM_CAPS
+		}
+
+		format s16le
+		index $HDMI1_HOST_PIPELINE_ID
+	}
+	passthrough-be.20 {
+		direction	"playback"
+		index $HDMI1_DAI_PIPELINE_ID
+		Object.Widget.pipeline.1 {
+			stream_name $HDMI1_PIPELINE_STREAM_NAME
+		}
+
+		Object.Widget.copier.1 {
+			stream_name "iDisp1"
+			dai_type "HDA"
+			copier_type "HDA"
+		}
+		format s16le
+	}
+	passthrough-playback.21 {
+		Object.Widget.pipeline.1 {
+			stream_name $HDMI2_PIPELINE_STREAM_NAME
+		}
+		Object.Widget.copier.1 {
+			stream_name $HDMI2_PCM_CAPS
+		}
+
+		format s16le
+		index $HDMI2_HOST_PIPELINE_ID
+	}
+	passthrough-be.21 {
+		direction	"playback"
+		index $HDMI2_DAI_PIPELINE_ID
+		Object.Widget.pipeline.1 {
+			stream_name $HDMI2_PIPELINE_STREAM_NAME
+		}
+
+		Object.Widget.copier.1 {
+			stream_name 'iDisp2'
+			dai_type "HDA"
+			copier_type "HDA"
+		}
+		format s16le
+	}
+	passthrough-playback.22 {
+		Object.Widget.pipeline.1 {
+			stream_name $HDMI3_PIPELINE_STREAM_NAME
+		}
+		Object.Widget.copier.1 {
+			stream_name $HDMI3_PCM_CAPS
+		}
+
+		format s16le
+		index $HDMI3_HOST_PIPELINE_ID
+	}
+	passthrough-be.22 {
+		direction	"playback"
+		index $HDMI3_DAI_PIPELINE_ID
+		Object.Widget.pipeline.1 {
+			stream_name $HDMI3_PIPELINE_STREAM_NAME
+		}
+
+		Object.Widget.copier.1 {
+			stream_name 'iDisp3'
+			dai_type "HDA"
+			copier_type "HDA"
+		}
+		format s16le
+	}
+}
+Object.PCM {
+	pcm.5 {
+		name HDMI1
+		id 5
+		Object.Base.fe_dai.HDMI1 {}
+		Object.PCM.pcm_caps.playback {
+			name $HDMI1_PCM_CAPS
+			formats 'S32_LE,S24_LE,S16_LE'
+		}
+		direction playback
+	}
+	pcm.6 {
+		name HDMI2
+		id 6
+		Object.Base.fe_dai.HDMI2 {}
+		Object.PCM.pcm_caps.playback {
+			name $HDMI2_PCM_CAPS
+			formats 'S32_LE,S24_LE,S16_LE'
+		}
+		direction playback
+	}
+	pcm.7 {
+		name HDMI3
+		id 7
+		Object.Base.fe_dai.HDMI3 {}
+		Object.PCM.pcm_caps.playback {
+			name $HDMI3_PCM_CAPS
+			formats 'S32_LE,S24_LE,S16_LE'
+		}
+		direction playback
+	}
+}
+
+# top-level pipeline connections
+Object.Base.route.10 {
+        source 'copier.host.5.1'
+        sink 'copier.HDA.6.1'
+}
+Object.Base.route.11 {
+        source 'copier.host.7.1'
+        sink 'copier.HDA.8.1'
+}
+Object.Base.route.12 {
+        source 'copier.host.9.1'
+        sink 'copier.HDA.10.1'
+}
+
+# include HDMI config if needed.
+IncludeByKey.NUM_HDMIS {
+"4" "platform/intel/hdmi-4th.conf"
```

Comment:
why does this need to be conditional at all? we plan to use topology2 for MTL+. This should be 4 always right?

Reply:
> why does this need to be conditional at all? we plan to use topology2 for MTL+. This should be 4 always right?

ok. I will use 4 hdmi unconditionally.

Reply:
> > why does this need to be conditional at all? we plan to use topology2 for MTL+. This should be 4 always right?
> 
> ok. I will use 4 hdmi unconditionally.

@ranj063 But skl_hda_dsp_generic machine driver only support 3 iDisp dai links, no matter what platform it is. So we probably need to support 3 HDMI?

Reply:
@ranj063 what prevents us from adding a 4th HDMI BE link in the machine driver? We just need to be careful with the BE index since there will be a discontinuity, but other than that it should just work fine.

Reply:
quick PR: https://github.com/thesofproject/linux/pull/3685

Reply:
As we discussed in https://github.com/thesofproject/linux/pull/3685, I will keep `NUM_HDMIS` and support different HDMI numbers.